types/react-router/test/WithRouter.tsx

```diff
@@ -22,3 +22,31 @@ const WithRouterTestFunction = () => (
     <WithRouterComponentFunction username="John" />
 );
 const WithRouterTestClass = () => <WithRouterComponentClass username="John" />;
+
+// union props
+{
+    interface Book {
+        kind: 'book';
+        author: string;
+    }
+
+    interface Magazine {
+        kind: 'magazine';
+        issue: number;
+    }
+
+    type SomethingToRead = (Book | Magazine) & RouteComponentProps;
+
+    const Readable: React.SFC<SomethingToRead> = props => {
```

Comment:
I think we might want to check which version of  `@types/react` is being used as I think `SFC` has been replaced by `FunctionComponent` since 16.8.

Reply:
It's deprecated. However it was already being used in the tests so changing it here wouldn't be right. At one point someone has to replace all the deprecated types.